py/jpy-integration/build.gradle

```diff
@@ -82,54 +83,129 @@ idea {
-      check.dependsOn t
-      pythonTests += t
-  }
+tasks.create 'pathJavaToPython', {
+  Task t ->
+    t.group = 'python'
+    t.description = 'Run the java exec io.deephaven.jpy.integration.PySysPath which prints java properties and python paths'
+}
 
-  tasks.create 'debugJavaToPython', {
-    Task t ->
-      t.group = 'python'
-      t.description = 'Run the java exec io.deephaven.jpy.integration.PyDebug which is a sanity test for all JavaExec tasks'
-  }
+Task pythonToJava = tasks.create 'unittestPythonToJava', {
+  Task t ->
+    t.group = 'python'
+    t.description = 'Run "python -m xmlrunner discover" in src/python-to-java/python'
+    check.dependsOn t
+}
 
-  tasks.create 'pathJavaToPython', {
-    Task t ->
-      t.group = 'python'
-      t.description = 'Run the java exec io.deephaven.jpy.integration.PySysPath which prints java properties and python paths'
-  }
+if (!PyEnv.pythonEnabled(project)) {
+//  Task testTask = venvTest.javaTest(project, "java-to-python-test-${pv.name}", installWheels, sourceSets.javaToPython)
+//  testJavaToPython.dependsOn testTask
+
+  Closure<TaskProvider<Task>> javaMainInDocker = { String taskName, String javaMain, SourceSet sourceSet, List<String> jvmArgs = [] ->
+    return Docker.registerDockerTask(project, taskName) {
+      copyIn {
+        from(sourceSet.runtimeClasspath) {
+          into 'classpath'
+        }
+      }
+      parentContainers = [project(':Integrations').tasks.findByName('buildDeephavenPython')] // deephaven/runtime-base
+      imageName = 'deephaven/jpy-integration-java-to-python-tests'
+      dockerfile {
+        from 'deephaven/runtime-base'
 
-  tasks.create 'debugPythonToJava', {
-    Task t ->
-      t.group = 'python'
-      t.description = 'Run the python exec python/java_debug.py which is a sanity test for all python Exec tasks in this project'
+        copyFile 'classpath', '/classpath'
+      }
+      entrypoint = ['java', '-cp', '/classpath:/classpath/*',
+                    '-Djpy.jpyLib=/usr/local/lib/python3.7/dist-packages/jpy.cpython-37m-x86_64-linux-gnu.so',
+                    '-Djpy.jdlLib=/usr/local/lib/python3.7/dist-packages/jdl.cpython-37m-x86_64-linux-gnu.so',
+                    '-Djpy.pythonLib=/usr/lib/x86_64-linux-gnu/libpython3.7m.so.1.0'
+      ] + jvmArgs + javaMain
+
+      showLogsOnSuccess = true
+    }
   }
 
-  Task pythonToJava = tasks.create 'unittestPythonToJava', {
-    Task t ->
-      t.group = 'python'
-      t.description = 'Run "python -m xmlrunner discover" in src/python-to-java/python'
-      check.dependsOn t
-      pythonTests += t
+  debugJavaToPython.dependsOn javaMainInDocker('java-to-python-debug', 'io.deephaven.jpy.integration.PyDebug', sourceSets.javaToPython, ["-Djpy.debug=true"])
+
+  pathJavaToPython.dependsOn javaMainInDocker('java-to-python-path', 'io.deephaven.jpy.integration.PySysPath', sourceSets.javaToPython)
```

Comment:
Does it make sense to declare these dependencies in the original task creation, as opposed to after-the-fact?

Reply:
Probably, unless we ever think about more than one py that we test on - agreed that for dh integration we don't want that, but for jpy we might?